src/clients/TorrentClient.ts

```diff
@@ -1,25 +1,51 @@
 import { Metafile } from "../parseTorrent.js";
-import { DecisionAnyMatch, InjectionResult } from "../constants.js";
+import {
+	Decision,
+	DecisionAnyMatch,
+	InjectionResult,
+	VIDEO_DISC_EXTENSIONS,
+} from "../constants.js";
 import { getRuntimeConfig } from "../runtimeConfig.js";
 import { Searchee, SearcheeWithInfoHash } from "../searchee.js";
 import QBittorrent from "./QBittorrent.js";
 import RTorrent from "./RTorrent.js";
 import Transmission from "./Transmission.js";
 import Deluge from "./Deluge.js";
 import { Result } from "../Result.js";
+import { Label } from "../logger.js";
+import { hasExt } from "../utils.js";
 
-let activeClient: TorrentClient;
+let activeClient: TorrentClient | null = null;
+
+export type TorrentClientType =
+	| Label.QBITTORRENT
+	| Label.RTORRENT
+	| Label.TRANSMISSION
+	| Label.DELUGE;
+
+export interface GenericTorrentInfo {
+	infoHash: string;
+	category: string;
+	tags: string[];
+	trackers?: string[][];
+}
 
 export interface TorrentClient {
+	type: TorrentClientType;
 	isTorrentComplete: (
 		infoHash: string,
 	) => Promise<Result<boolean, "NOT_FOUND">>;
+	getAllTorrents: () => Promise<GenericTorrentInfo[]>;
 	getDownloadDir: (
 		meta: SearcheeWithInfoHash | Metafile,
 		options: { onlyCompleted: boolean },
 	) => Promise<
 		Result<string, "NOT_FOUND" | "TORRENT_NOT_COMPLETE" | "UNKNOWN_ERROR">
 	>;
+	getAllDownloadDirs: (options: {
+		metas: SearcheeWithInfoHash[] | Metafile[];
+		onlyCompleted: boolean;
+	}) => Promise<Map<string, string>>;
```

Comment:
I'm assuming this maps info hash to download dir?

Reply:
Yes. It works similar to `getAllTorrents()` but that has the infoHash in the object.